support iec eip resource and docs

Lance52259 opened the pull request:
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccIecEIPResource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccIecEIPResource_basic -timeout 360m -parallel 4
=== RUN   TestAccIecEIPResource_basic
=== PAUSE TestAccIecEIPResource_basic
=== CONT  TestAccIecEIPResource_basic
--- PASS: TestAccIecEIPResource_basic (37.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       37.259s

Lance52259 committed:
support iec eip resource and docs

ShiChangkuo commented on huaweicloud/resource_huaweicloud_iec_eip.go:
```diff
@@ -0,0 +1,287 @@
+	createOpts := publicips.CreateOpts{
+		Publicip: publicips.PublicIPRequest{
+			SiteID:    d.Get("site_id").(string),
+			IPVersion: strconv.Itoa(d.Get("ip_version").(int)),
```
set the IPVersion when the `ip_version` is not zero

ShiChangkuo commented on huaweicloud/resource_huaweicloud_iec_eip.go:
```diff
@@ -0,0 +1,287 @@
+	_, err := publicips.Update(client, d.Id(), updateOpts).Extract()
+	if err != nil {
+		var action string = "binding"
+		if port != "" {
```
should be `port == ""`

ShiChangkuo commented on huaweicloud/resource_huaweicloud_iec_eip.go:
```diff
@@ -0,0 +1,287 @@
+			"ip_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntInSlice([]int{4}),
```
suggest to add `Computed`

ShiChangkuo commented on docs/resources/iec_eip.md:
```diff
@@ -0,0 +1,66 @@
+* `site_id` - (Required, String, ForceNew) Specifies the id of iec sevice site. 
+    Changing this parameter creates a new iec eip resource.
+
+* `ip_version` - (Optional, Int) Specifies the version of elastic IP address. 
```
as the value only can be 4, suggest to move the block to `Attribute Reference`

ShiChangkuo commented on docs/resources/iec_eip.md:
```diff
@@ -0,0 +1,66 @@
+
+* `id` - The resource ID in UUID format.
+
+* `status` - Specifies the [status](https://support.huaweicloud.com/intl/en-us/api-eip/eip_api_0002.html#eip_api_0002__en-us_topic_0201534285_table3035698) 
```
the hyperlink is status of vpc eip rather than iec eip

Lance52259 committed:
Update resource_huaweicloud_iec_eip.go

1. Update ip_version to Computed mode.
2. Set ip_version as IPv4(4) only it's empty.  
3. set action as "unbinding" when port is null.

Lance52259 committed:
Update iec_eip.md

change list:
1. move ip_version from Argument Reference to Attributes Reference.

Lance52259 committed:
Update iec_eip.md

Remove status hyperlink of iec eip.

Lance52259 committed:
Update iec_eip.md

Lance52259 committed:
Update resource_huaweicloud_iec_eip_test.go

remove ip_version check.

Lance52259 committed:
Update resource_huaweicloud_iec_eip_test.go

Update function of parameter check.
